README.txt
FRAME3DD: a frame analysis program
-------------------------------

FRAME3DD is a program for the static and dynamic structural analysis of
two- and three-dimensional frames and trusses with elastic and 
geometric stiffness.

FRAME3DD reads an Input Data file, containing joint coordinates, member
geometry, material moduli, restrained joints, prescribed displacements,
load information, and optionally, mass information if a modal analysis
is to be carried out.

FRAME3DD appends results to an Output Data File, The Output Data 
recapitulates the input information, gives joint displacements in global 
coordinates, member end-forces in local coordinates, reactions in 
global coordinates, and natural frequencies and mode shapes in global 
coordinates.

For usage information, see the User Manual, included with both the source
code and binary installer packages.

Additional output files from FRAME3DD are directed by default to the temporary
file folder (according to the %TEMP% env var on Windows, or /tmp on other
platforms). You can specify an alternative location for output files by setting
the environment variable FRAME3DD_OUTDIR.

Note that you require write permissions for the input file in order that
FRAME3DD can append its calculation results to that file.

---

FRAME3DD is free software; you may redistribute it and/or modify it under 
the terms of the GNU General Public License (GPL) as published by the
Free Software Foundation. FRAME3DD is distributed in the hope that it will
be useful, but WITHOUT ANY WARRANTY; without even the implied warranty
of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
General Public License (GPL) for details. 

---

You should have obtained this distribution from SourceForge.net 

svn co https://frame3dd.svn.sourceforge.net/svnroot/frame3dd frame3dd 

---

Included in this distribution:

LICENSE.txt  ... GNU GPL license
README.txt   ... this file

SConstruct   ... optionally used for compiling and building distribution
                 using scons ... http://www.scons.org/
frame3dd.spec... build specification for scons, includes changelog
save.plt     ... Gnuplot macro for saving Gnuplot plots as PostScript

doc/         ... documentation (in HTML format)
examples/    ... examples of FRAME input/output files
scons/       ... used by scons to build distribution
src/         ... C source code 

to compile and build without scons ...

gcc -O -Wall -o frame3dd main.c frame3dd.c frame3dd_io.c ldl_dcmp.c lu_dcmp.c coordtrans.c eig.c hpgUtils.c nrutil.c -lm

---

FRAME3DD author

(c) 1992-2008 Henri Gavin - Associate Professor 
Department of Civil and Environmental Engineering
Edmund T. Pratt School of Engineering
Duke University - Box 90287, Durham, NC 27708-0287

mailto:Henri.Gavin@Duke.edu - tel: 919-660-5201 - fax: 919-660-5219 

---

Packaging of FRAME3DD for distribution on Linux and Windows,
as well as implementation of the Microstran file parsers
is by John Pye, 2007-2013.

John Pye
Department of Engineering
Australian National University
mailto:john.pye@anu.edu.au
http://jpye.dyndns.org/

---



ChangeLog.txt
* Wed Nov 25 14:06:00 EST 2015 Henri Gavin <hpgavin> 0.20140514+
- changed Matlab interface function frame_3dd.m so that 
floating point values in the .FMM file have aligned decimal points.

* Sun Oct 31 15:15:00 EST 2015 Henri Gavin <hpgavin> 0.20140514+
- add the peak element internal forces to the .CSV output data file
- update Frame3DD-manual.html re. the inclusion of peak element internal
forces in the .CSV output data file

* Tue Oct 20 15:53:00 EST 2015 Henri Gavin <hpgavin> 0.20140514+
- in testing for large average strains, check for large positive _and_
negative strains 
frame3dd.c line 450 changed ... 
if ( fabs(axial_strain > 0.001 ) ) {
... to ... 
if ( fabs(axial_strain) > 0.001 ) {
- new Exit Codes:
181 : stiffness matrix not positive definite
182 : large strain
183 : stiffness matrix not positive definite _and_ large strains
- Frame3DD-manual.html updated with new Exit Codes

* Fri Jun  5 16:33:00 EST 2015 Henri Gavin <hpgavin> 0.20140514+
- changed format of the .zip file distributions.  now all binary applications
are in a single distribution .zip file.   
- change installation instructions to comply with the new distribution file
structure
- change sourceforge location of download .zip archive to
https://sourceforge.net/projects/frame3dd/files/
- did not change the link used to the download .zip archive ... still ...
http://sourceforge.net/project/platformdownload.php?group_id=221386
... this link finds the latest .zip file in the archive.
... not yet sure if the download stats are updating correctly.
- small style formatting changes to user manual


* Thu Jun  4 16:33:00 EST 2015 Henri Gavin <hpgavin> 0.20140514+
- increase limit on number of animated modes from 20 to 100.
- function name changes in main.c frame3dd_io.c frame3dd.c ...
-- rename conden to static_condensation
-- rename guyan to paz_condensation
-- rename dynamic_conden to modal_condensation
- associated updates to user manual
- replace links with platformdownload in the user manual with links to list of
download .zip files ... the platformdownload option at sourceforge appears to
be no longer implemented.


* Fri Dec  5 13:11:00 EST 2014 Henri Gavin <hpgavin> 0.20140514+
- require "anlyz ~= 0" in order to run matrix condensation


* Fri Nov  7 13:11:00 EST 2014 Henri Gavin <hpgavin> 0.20140514+
- update Frame3DD-manual.html with instructions on how to zoom, pan, and shift
3D images in gnuplot.
- update saveplot and saveplot_w32 for gnuplot version 4.6


* Sat May 17 07:00:00 EST 2014 Henri Gavin <hpgavin> 0.20140514
- clear a few warnings when compiling with -Wall
- add more precision to the description of the numerical method in the
user-manual.html
- rename user-manual.html to Frame3DD-manual.html
- fix link in website/index.html
- add Frame3DD-manual.html to sourceforge repository and distributions
- add scale variable to Input Data File to scale 3D plots in gnuplot 4.6

* Fri May 16 07:00:00 EST 2014 Henri Gavin <hpgavin> 0.20140514
- Reaction forces reflect the presence of loads applied directly on reaction coordinates
- Reactions are computed and updating for partitioned matrix systems
within ldl_dcmp_pm and ldl_mprove_pm. computing reactions in this way
is valid only for linear elastic analyses
- function compute_reaction_forces() was re-introduced ... 
The superposition of incremental reaction forces dR does not 
work in problems with geometric stiffness, since
the stiffness changes at each iteration.    
So the reactions are re-computed only for problems with geometric stiffness.    
- Compute RMS equilibrium error for linear elastic analyses
- The RMS equilibrium error makes use of only the upper triangle of [K_qq]
- RMS equilibrium values are assessed for linear elastic analyses

* Thu May 15 07:00:00 EST 2014 Henri Gavin <hpgavin> 0.20140514
- merge add_feF() into element_end_forces()

* Wed May 14 07:00:00 EST 2014 Henri Gavin <hpgavin> 0.20140514
- include the effect of applied point loads at reaction coordinates on the computed reaction forces
This change has simplified the calculations by eliminating the need for 
the entire function: compute_reaction_forces() and 
the reaction-related calcluations in the function: add_feF().   
Instead, the correct reaction forces are now computed within ldl_dcmp_pm.    
The relevant changes to main.c, frame3dd.c, and HPGmatrix.c are 
commented with the date of the change: 2014-05-14.
Because this change affects computed results,
the version number is updated to 20140514.
Step 8 of section 8 of the user manual is updated to reflect
this change in the method.
- these changes do not affect the formatting of the input or output data
- changed all gnuplot commands "set nofoobar" to "unset foobar"
- caught and fixed bug affecting smoothness of mode-shape animations
- updated links to ansicon and Sketchup interface in user-manual.html

* Fri Jan 24 08:00:00 EST 2014 Henri Gavin <hpgavin> 0.20140121
- update zipdist.sh for OSX versions "789"

* Tue Jan 21 16:00:00 EST 2014 Henri Gavin <hpgavin> 0.20140121
- correction in computation of internal element transverse displacements
- apply linear correction to integrated (M/EI) *before* adding in V/GA
- update version numbers to 20140121 in code and documentation

* Mon Dec 30 15:58:18 EST 2013 Henri Gavin <hpgavin> 0.20130317
- sign of reaction forces in .CSV output is negative of what it should be.
- source-code and examples updated

*Thu Dec 19 22:19:21 EST 2013 Henri Gavin <hpgavin> 0.20130317
- add .csv versions of all examples to web site and to download collection
- remove semicolon (;) from list of comment delimiters for "csv" compatability
- add semicolon (;) to list of ignored characters for "csv" compatability

*Tue Dec 17 22:19:21 EST 2013 Henri Gavin <hpgavin> 0.20130317
- user_manual.html ... change .bash_profile to .profile

*Mon Dec 16 22:19:21 EST 2013 Henri Gavin <hpgavin> 0.20130317
- user_manual.html ... change .bashrc to .bash_profile
- user_manual.html ... change .cshrc  to .login


zipdist.sh
#!/bin/bash
#
# zipdist.sh
# assemble .ZIP files for Frame3dd executable and source distribution release

# NOTE: this version number should preferably be synchronised with the version
# number in file 'SConstruct'. Note the alternative packaging instructions in 
# the file 'PACKAGING.txt' which builds DEB, RPM and EXE packages, but nothing
# for Mac users :-( 
export VERSION=20140514+
echo $VERSION

# clean out prior distribution files
echo "cleaning out prior distribution files ... "
rm -rf dist/Frame3DD

# make directory structure for distribution .ZIP file in the ./dist directory
echo "creating directory structure for new distribution ... "
mkdir dist/Frame3DD
mkdir dist/Frame3DD/doc
mkdir dist/Frame3DD/doc/img
mkdir dist/Frame3DD/examples
mkdir dist/Frame3DD/matlab
mkdir dist/Frame3DD/linux
mkdir dist/Frame3DD/osx
mkdir dist/Frame3DD/windows

# make an empty temp directory
echo "making an empty Frame3DD/temp directory ... "
mkdir dist/Frame3DD/temp/

echo "creating source code directory structure for new distribution ... "
mkdir dist/Frame3DD/src
mkdir dist/Frame3DD/src/microstran

# copy documentation
echo "copying documentation ... "
cp --preserve=mode,timestamps doc/*.html            dist/Frame3DD/doc/.
cp --preserve=mode,timestamps doc/template.3dd      dist/Frame3DD/doc/.
cp --preserve=mode,timestamps doc/template.csv      dist/Frame3DD/doc/.
cp --preserve=mode,timestamps doc/img/*.png         dist/Frame3DD/doc/img/.
cp --preserve=mode,timestamps doc/img/*.jpg         dist/Frame3DD/doc/img/.
cp --preserve=mode,timestamps doc/img/*.gif         dist/Frame3DD/doc/img/.
#cp --preserve=mode,timestamps doc/LinuxTerminal.pdf dist/Frame3DD/doc/.	

# copy ChangeLog, license, and README files
cp --preserve=mode,timestamps ChangeLog.txt         dist/Frame3DD/.
cp --preserve=mode,timestamps LICENSE.txt           dist/Frame3DD/.
cp --preserve=mode,timestamps README.txt            dist/Frame3DD/.
cp --preserve=mode,timestamps README-win32.txt      dist/Frame3DD/.

# copy example files (text and .csv formats)
echo "copying example files ... "
cp --preserve=mode,timestamps examples/ex*.3dd      dist/Frame3DD/examples/.
cp --preserve=mode,timestamps examples/ex*.out      dist/Frame3DD/examples/.
cp --preserve=mode,timestamps examples/ex*.csv      dist/Frame3DD/examples/.
cp --preserve=mode,timestamps examples/ex*_out.CSV  dist/Frame3DD/examples/.
cp --preserve=mode,timestamps examples/saveplot     dist/Frame3DD/examples/.
cp --preserve=mode,timestamps examples/saveplot_w32 dist/Frame3DD/examples/.

# copy matlab files
echo "copying matlab files ... "
cp --preserve=mode,timestamps matlab/*              dist/Frame3DD/matlab/.

# copy source code
echo "copying source code ... "
cp --preserve=mode,timestamps src/*.c               dist/Frame3DD/src/.
cp --preserve=mode,timestamps src/*.h               dist/Frame3DD/src/.
cp --preserve=mode,timestamps src/Makefile          dist/Frame3DD/src/.
cp --preserve=mode,timestamps src/microstran/vec3.h dist/Frame3DD/src/microstran/.
cp --preserve=mode,timestamps src/microstran/config.h dist/Frame3DD/src/microstran/.


# copy binary executable application
echo "copying Linux executable ... "
cp --preserve=mode,timestamps build/linux/frame3dd        dist/Frame3DD/linux/frame3dd
echo "copying OS X executable ... "
cp --preserve=mode,timestamps build/osx/frame3dd          dist/Frame3DD/osx/frame3dd
echo "copying Windows executable ... "
cp --preserve=mode,timestamps build/windows/frame3dd.exe  dist/Frame3DD/windows/frame3dd.exe
cp --preserve=mode,timestamps build/DukeOIT/frame3dd      dist/Frame3DD/DukeOIT/frame3dd

# assemble the .zip file
echo "assembling .zip file ... "
cd dist                                           # change to trunk/dist
zip -r Frame3DD_$(echo $VERSION).zip Frame3DD/*
#zip -r Frame3DD_$(date +%Y%m%d).zip Frame3DD/*

#rm Frame3DD_$(date +%Y%m%d).zip

echo "Frame3DD .zip archive complete"

rm -rf Frame3DD

# To upload distribution files to sourceforge.net ...
#
# rsync -uav Frame3DD*.zip hpgavin,frame3dd@frs.sourceforge.net:/home/frs/project/f/fr/frame3dd/.
#
##rsync -uav *.bz2  hpgavin,frame3dd@frs.sourceforge.net:/home/frs/project/f/fr/frame3dd/frame3dd/0.20140514+/.

# ----------------------------------------------------------------- zipdist.sh
# Henri P. Gavin  2009-10-20
# updated: 2009-10-22, 2009-10-25, 2009-10-27, 2009-10-29, 2010-01-05, 2010-12-1, 2013-03-18, 2014-01-21, 2014-05-14, 2014-05-17, 2015-06-04
